Fixed missing visit functions in ASTVisitor

jenrik opened the pull request:
Added new visit functions for FunctionArgs, Coordinate, WorldNode,
WorldDimension, ForStmt, BreakStmt, and ContinueStmt to ASTVisitor.
Added missing visit funtion to BaseASTVisitor for WorldNode,
WorldDimension, FunctionArgs, FloatLiteral, Coordinate, ForStmt,
BreakStmt, and ContinueStmt.
Fixed BaseASTVisitor#visit(NeighbourhoodDecl) not visiting coordinates.
Added value extraction for WorldNode and Coordinate.
Added missing body of "for" statement to ForStmt.
Added missing reduce call for "for" statement body.

jenrik committed:
Fixed missing visit functions in ASTVisitor

Added new visit functions for FunctionArgs, Coordinate, WorldNode,
WorldDimension, ForStmt, BreakStmt, and ContinueStmt to ASTVisitor.
Added missing visit funtion to BaseASTVisitor for WorldNode,
WorldDimension, FunctionArgs, FloatLiteral, Coordinate, ForStmt,
BreakStmt, and ContinueStmt.
Fixed BaseASTVisitor#visit(NeighbourhoodDecl) not visiting coordinates.
Added value extraction for WorldNode and Coordinate.
Added missing body of "for" statement to ForStmt.
Added missing reduce call for "for" statement body.

falkecarlsen commented on compiler/src/main/kotlin/walkers/visitor.kt:
```diff
@@ -337,24 +370,43 @@ abstract class BaseASTVisitor: ASTVisitor {
 
     override fun visit(node: PreIncStmt) {
-        // no-op
+        visit(node.variable)
```
Remember that all pre/post, inc/dec statements have been removed. These are not present in neither the grammar nor compiler source on master.

falkecarlsen commented on compiler/src/main/kotlin/walkers/value.kt:
```diff
@@ -152,4 +153,45 @@ class LiteralExtractorVisitor : BaseASTVisitor() {
+        }
+    }
+
+    override fun visit(node: WorldNode) {
```
Please add some meaningful documentation to the following function.

falkecarlsen reviewed (dismissed):
Looks good, but there's an issue with your feature-branch having pre/post, inc/dec statements still implemented although master does not have these.

jenrik committed:
Documented new functions in valueextractor.kt

Documented LiteralExtractorVisitor#toInt.
Documented the internals of LiteralExtractorVisitor#visit(WorldNode).

falkecarlsen commented on compiler/src/main/kotlin/visitors/valueextractor.kt:
```diff
@@ -152,4 +153,54 @@ class LiteralExtractorVisitor : BaseASTVisitor() {
+
+        val width = parseDimension(node.ctx.size.width)
+
+        // If height is non-null their is two dimension, otherwise their is only one
```
```suggestion
        // If height is non-null it is two dimensional, otherwise it is one dimensional
```

NicEastvillage replied:
```suggestion
        // If height is non-null there are two dimensions, otherwise there is only one
```

falkecarlsen commented on compiler/src/main/kotlin/visitors/valueextractor.kt:
```diff
@@ -152,4 +153,54 @@ class LiteralExtractorVisitor : BaseASTVisitor() {
+                type = when(type) {
+                    is CellmataParser.DimFiniteEdgeContext -> WorldType.EDGE
+                    is CellmataParser.DimFiniteWrappingContext -> WorldType.WRAPPING
+                    else -> throw AssertionError()
```
Should this not utilise our newly created ErrorLogger?

jenrik replied:
I don't think so, this is a case where if we hit the else statements it is not the user's fault, it's because the compiler is broken.

falkecarlsen replied:
Okay, cool

NicEastvillage commented on compiler/src/main/kotlin/visitors/valueextractor.kt:
```diff
@@ -152,4 +153,54 @@ class LiteralExtractorVisitor : BaseASTVisitor() {
     }
+
+    /**
+     * Parse a string into a integer, or throw IntegerParisingException the string isn't a valid integer.
```
```suggestion
     * Parse a string into a integer, or throw IntegerParsingException the string isn't a valid integer.
```

falkecarlsen reviewed (changes requested):
One question and one spelling-suggestion

jenrik committed:
Apply suggestions from code review

Co-Authored-By: jenrik <jenrik@users.noreply.github.com>
